src/Donations/Controllers/DonationRequestController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Give\Donations\Controllers;
+
+use Give\Donations\Models\Donation;
+use Give\Donations\Repositories\DonationRepository;
+use Give\Donations\ValueObjects\DonationRoute;
+use Give\Framework\QueryBuilder\JoinQueryBuilder;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+
+/**
+ * @unreleased
+ */
+class DonationRequestController
+{
+    /**
+     * @unreleased
+     *
+     * @return WP_Error | WP_REST_Response
+     */
+    public function getDonation(WP_REST_Request $request)
+    {
+        $donation = Donation::find($request->get_param('id'));
+
+        if ( ! $donation) {
+            return new WP_Error('donation_not_found', __('Donation not found', 'give'), ['status' => 404]);
+        }
+
+        return new WP_REST_Response($donation->toArray());
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getDonations(WP_REST_Request $request): WP_REST_Response
+    {
+        $page = $request->get_param('page');
+        $perPage = $request->get_param('per_page');
+
+        $query = give(DonationRepository::class)->prepareQuery();
+
+        if ($campaignId = $request->get_param('campaignId')) {
+            $query->distinct()->join(function (JoinQueryBuilder $builder) {
```

Comment:
This should probably use the `CampaignDonationQuery`, which would cover subscription donations and account for the payment mode (live vs test).

Reply:
Since this endpoint is related to the `Donation` resource, I would prefer to use the `Donation::query()` to generate the base query and then add the proper where conditions and filters attached to it. Even because the $campaingId parameter is optional.

Resolved in 7a086f52fdae9c9df1da630d9967e10931ef4d15 and 4fca5af2d7ecc1665e6f4f951c01efca53652b02